test/composableStableJoin.integration.test.ts

```diff
@@ -0,0 +1,309 @@
+// pnpm test -- composableStableJoin.integration.test.ts
+import { describe, expect, test, beforeAll, beforeEach } from 'vitest';
+import { config } from 'dotenv';
+config();
+
+import {
+    createTestClient,
+    http,
+    parseUnits,
+    publicActions,
+    walletActions,
+} from 'viem';
+
+import {
+    UnbalancedJoinInput,
+    ProportionalJoinInput,
+    SingleAssetJoinInput,
+    JoinKind,
+    Slippage,
+    Token,
+    TokenAmount,
+    Address,
+    Hex,
+    PoolStateInput,
+    CHAINS,
+    ChainId,
+    getPoolAddress,
+    PoolJoin,
+    JoinInput,
+} from '../src';
+import { forkSetup } from './lib/utils/helper';
+import { JoinTxInput } from './lib/utils/types';
+import { doJoin } from './lib/utils/joinHelper';
+
+const chainId = ChainId.MAINNET;
+const rpcUrl = 'http://127.0.0.1:8545/';
+const poolId =
+    '0x156c02f3f7fef64a3a9d80ccf7085f23cce91d76000000000000000000000570'; // Balancer vETH/WETH StablePool
+
+describe('composable stable join test', () => {
+    let txInput: JoinTxInput;
+    let bptToken: Token;
+
+    beforeAll(async () => {
+        // setup mock api
+        const api = new MockApi();
+
+        // get pool state from api
+        const poolInput = await api.getPool(poolId);
+
+        const client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+
+        txInput = {
+            client,
+            poolJoin: new PoolJoin(),
+            slippage: Slippage.fromPercentage('1'), // 1%
+            poolInput,
+            testAddress: '0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f', // Balancer DAO Multisig
+            joinInput: {} as JoinInput,
+            checkNativeBalance: false,
+            chainId,
+        };
+
+        // setup BPT token
+        bptToken = new Token(chainId, poolInput.address, 18, 'BPT');
+    });
+
+    beforeEach(async () => {
+        await forkSetup(
+            txInput.client,
+            txInput.testAddress,
+            [...txInput.poolInput.tokens.map((t) => t.address)],
+            [0, 0, 3],
+            [
+                ...txInput.poolInput.tokens.map((t) =>
+                    parseUnits('100', t.decimals),
+                ),
+            ]
+        );
+    });
+
+    test('unbalanced join', async () => {
+        const bptIndex = txInput.poolInput.tokens.findIndex(
+            (t) => t.address === txInput.poolInput.address,
+        );
+        const poolTokensWithoutBpt = txInput.poolInput.tokens
+            .map((t) => new Token(chainId, t.address, t.decimals))
+            .filter((_, index) => index !== bptIndex);
+
+        const amountsIn = poolTokensWithoutBpt.map((t) =>
+            TokenAmount.fromHumanAmount(t, '1'),
+        );
```

Comment:
Should we go ahead and update `amountsIn` to minimal token info (address + rawAmount)? Or is this better suited for a separate refactor PR that will cover all regular joins? @johngrantuk

Reply:
Yeah, I was thinking this can be done in a later refactor after this is merged. I've made an Asana already.